domains/auth/Auth.API/API/Services/CryptographyService.cs

```diff
@@ -0,0 +1,56 @@
+using API.Models;
+using API.Helpers;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace API.Services;
+public class CryptographyService : ICryptographyService
+{
+    string key = Configuration.GetSecretKey();
+
+    public string EncryptState(AuthState state)
+    {
+        byte[] iv = new byte[16];
+        using (Aes aes = Aes.Create())
+        {
+            aes.Key = Encoding.UTF8.GetBytes(key);
+            aes.IV = iv;
+            ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
+            using (MemoryStream memoryStream = new MemoryStream())
+            {
+                using (CryptoStream cryptoStream = new CryptoStream(memoryStream, encryptor, CryptoStreamMode.Write))
+                {
+                    using (StreamWriter streamWriter = new StreamWriter(cryptoStream))
+                    {
+                        streamWriter.Write(state.ToString());
+                    }
+                    byte[] data = memoryStream.ToArray();
+                    return Convert.ToBase64String(data);
+                }
```

Comment:
```suggestion
                }
                byte[] data = memoryStream.ToArray();
                return Convert.ToBase64String(data);
```
i would suggest moving this outside the cryptostream.